### PR TITLE
Improves Login flow

### PR DESCRIPTION
**Pull request by saketkumar95**
Fixes https://github.com/zulip/zulip-mobile/issues/456

- [x] Register and forgot password option

### CODE REVIEW

**Comment by saketkumar95** [comment]
@borisyankov Done with the register and forgot password option. For intro tutorials, PR https://github.com/zulip/zulip-mobile/pull/468 already exists. Is that PR active or should i implement the tutorial sliders?

**Comment by saketkumar95** [comment]
@borisyankov @timabbott Here are the screenshots of the new design.

**Comment by borisyankov** [comment]
Try showing the keyboard and see if the screen designs steel work.

**Comment by borisyankov** [comment]
Also, skip 'Intro Sliders' and work on #475 in a separate PR.

**Comment by saketkumar95** [comment]
@borisyankov Here is how screen looks when keyboard is up!

**Comment by borisyankov** [comment]
Yes... now imagine this is the only screen they see.
On an actual phone we will not hide the keyboard.

**Comment by saketkumar95** [comment]
So should i resize the image or just remove it?

**Comment by timabbott** [comment]
I think we might want to use a different image.  I would recommend removing the image for now, merging the other improvements, and then we can work with @vaidap on coming up with a good illustration for this screen that's less tall than the balloon.

**Comment by vaidap** [comment]
this is looking so nice! i'd be up for doing the illustration. if you have any ideas for the concept, do let me know-- i'll sketch something by the end of next week. :)

**Comment by saketkumar95** [comment]
@timabbott @borisyankov Okay so, Currently i'm removing the image part. :)

**Comment by vaidap** [comment]
Something more horizontal that could work for logging in -- flock of fish boarding a bus! let me know your thoughts. alternatively, it could be used for a newsletter subscription headline :)
image

**Comment by saketkumar95** [comment]
@vaidap This really looks nice to be. My suggestion could be like having a animation of flock of fish boarding the bus and having this animation in the splash screen.
@borisyankov  @timabbott We don't have a splash screen right now. What do you think about this? Or we can add this animation in the login page.

**Comment by vaidap** [comment]
@saketkumar95 oh, the animation is a wonderful idea! is a GIF a good format for this?

**Comment by timabbott** [comment]
Yeah, that concept looks good!  Can you post it on #design for more feedback?  I could imagine adding a bit more details on the sea creatures floating after the bus, but that's detail work :)

I would consider going with a non-animated version for now, and plan to refine to add animations later.  Animations can take a lot of time to get right, and that could make sense to save for a v2.  But I'm interested in others' thoughts on this.

**Comment by saketkumar95** [comment]
@vaidap Yeah GIF format would be nice one. :)

**Comment by borisyankov** [comment]
Guys! Reality check. This is a solution to a problem we don't have!

In fact we have lots of good problems that can be solved with visuals. But let's start with the problem. Then come up with a solution.

**Comment by brockwhittaker** [comment]
@borisyankov we're currently doing designs for our emails right now which includes some graphics.

**Comment by saketkumar95** [comment]
@borisyankov I have updated the PR. 
This PR contains : 

 Register and forgot password option.
 Improves the textAlign.

for intro tutorial sliders, I'll be creating new PR.

File: src/start/PasswordAuthView.js [hunk 1]
```diff
@@ -90,6 +90,16 @@ class PasswordAuthView extends React.Component {
           onPress={this.validateForm}
         />
         <ErrorMsg error={error} />
+        <TouchableOpacity activeOpacity={1.5}>
+          <View>
+            <Text style={[styles.field, styles.common]} onPress={() => Linking.openURL('https://chat.zulip.org/accounts/password/reset/')}>Forgot Password? </Text>
```
Comment (vishwesh3) [inline src/start/PasswordAuthView.js]: we can not hard code URLs, it varies with different servers.

File: src/start/PasswordAuthView.js [hunk 2]
```diff
@@ -90,6 +90,16 @@ class PasswordAuthView extends React.Component {
...
+          </View>
+        </TouchableOpacity>
+        <TouchableOpacity activeOpacity={1.5}>
+          <View>
+            <Text style={[styles.field, styles.common]} onPress={() => Linking.openURL('https://chat.zulip.org/register/')}>Register </Text>
```
Comment (vishwesh3) [inline src/start/PasswordAuthView.js]: here too.

File: src/common/styles.js [hunk 3]
```diff
@@ -31,19 +31,27 @@ export default StyleSheet.create({
     alignItems: 'stretch',
     marginTop: STATUSBAR_HEIGHT,
   },
+  common: {
```
Comment (vishwesh3) [inline src/common/styles.js]: I think this should be renamed. As it is already in `/common`. Then again with name 'common' creates confusion.

File: src/start/PasswordAuthView.js [hunk 4]
```diff
@@ -90,6 +90,16 @@ class PasswordAuthView extends React.Component {
           onPress={this.validateForm}
         />
         <ErrorMsg error={error} />
+        <TouchableOpacity activeOpacity={1.5}>
+          <View>
```
Comment (vishwesh3) [inline src/start/PasswordAuthView.js]: this view is redundant.

File: src/start/PasswordAuthView.js [hunk 5]
```diff
@@ -90,6 +90,16 @@ class PasswordAuthView extends React.Component {
           onPress={this.validateForm}
         />
         <ErrorMsg error={error} />
+        <TouchableOpacity activeOpacity={1.5}>
```
Comment (vishwesh3) [inline src/start/PasswordAuthView.js]: We have our common `Touchable`  for the project. https://github.com/zulip/zulip-mobile/blob/master/src/common/Touchable.ios.js

**Comment by vaidap** [comment]
Just a heads-up if you wanted to see what the login screen could look with a gif, I've made a quick version:
loginanim

**Comment by saketkumar95** [comment]
Yeah.. @vaidap I was waiting for this only. 
Looks really cool.

**Comment by vaidap** [comment]
@saketkumar95 glad to hear! might make a couple more changes & will update you as i do so. i might need to learn how to animate SVGs in order to prevent pixellation-- or does it look okay on a mobile device?

**Comment by saketkumar95** [comment]
@vaidap Do send me the gif at saket.singh1410@gmail.com
Downloading from here may cause the quality of the gif. :)

**Comment by vaidap** [comment]
sent!

**Pull request closed without merging**